src/main/java/Dan/Parser.java

```diff
@@ -0,0 +1,57 @@
+package Dan;
+
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    private TaskList tasks;
+
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+    
+    public boolean Parse(String input) {
+        String action = input.split(" ")[0];
+        try {
+            switch (action) {
+            case "bye":
+                Ui.sayonara();
+                return true;
+
+            case "list":
+                tasks.showTasks();
+                break;
+
+            case "mark":
+                tasks.markTask(Integer.parseInt(input.split(" ")[1]));
+                break;
+
+            case "unmark":
+                tasks.unMarkTask(Integer.parseInt(input.split(" ")[1]));
+                break;
+
+            case "delete":
+                tasks.deleteTask(Integer.parseInt(input.split(" ")[1]));
+                break;
+
+            case "todo":
+                //fall through
+            case "deadline":
+                //fall through
+            case "event":
+                tasks.addTask(input);
+                break;
+
+            default:
+                throw new DanException("I don't really understand what do you mean by that...");
```

Comment:
Nice job following the coding standards for the switch clause!

Reply:
It's good that the case clauses are not indented, but I think there shouldn't be an empty line between cases as well? Maybe its for clarity, but from what I read in the style guide it seems like there shouldn't be the extra lines.

Reply:
Thanks for pointing this out! Will take note of this.